public/app/core/services/LivePerformance.ts

```diff
@@ -0,0 +1,114 @@
+import config from 'app/core/config';
+
+import { fromPairs, mapValues, sortBy } from 'lodash';
+
+type IntervalStats = {
+  time: number;
+  count: number;
+  min: number;
+  max: number;
+  avg: number;
+};
+
+type LivePerformanceOptions = { maxIntervalsToKeep: number; intervalDuration: number };
+
+export enum MeasurementName {
+  DataRenderDelay = 'DataRenderDelay',
+  DashboardRenderBudgetExceeded = 'DashboardRenderBudgetExceeded',
+}
+
+const measurementNames = Object.keys(MeasurementName);
+
+export class LivePerformance {
```

Comment:
Should this live in: `public/app/features/live/` rather than services?   

Then the CODEOWNERS will point to edge

Reply:
sure,d one